### PR TITLE
Handle no internet connection gracefully

### PR DESCRIPTION
**Pull request by svartkanin**
When starting archinstall and there's not internet access it will fail with a cryptic error message such as reported in https://github.com/archlinux/archinstall/issues/1357 and https://github.com/archlinux/archinstall/issues/1347. 

# Changes
This PR validates if there's an internet connection available on the system and if not will terminate gracefully with a respective message 

```
Your system must have internet access to be able to run archinstall
```

### CODE REVIEW

**Comment by dylanmtaylor** [comment]
We want to be able to handle offline installations with local mirrors

File: archinstall/__init__.py [hunk 1]
```diff
@@ -245,6 +246,20 @@ def post_process_arguments(arguments):
 	load_config()
 
 
+def verify_internet_connection():
+	from urllib.request import urlopen
+	try:
+		import urllib.request
+		url = 'https://1.1.1.1'
```
Comment (dylanmtaylor) [inline archinstall/__init__.py:253]: Instead of cloudflare, we should be checking an activity to the actual mirrors

**Comment by dylanmtaylor** [comment]
We already tried to do a check for internet connectivity in guided with the line

 if not (archinstall.check_mirror_reachable() or archinstall.arguments.get('skip-mirror-check', False)):

Perhaps updating the message that is output if this does not succeed would be a better solution.

**Comment by svartkanin** [comment]
That's a good point. I'll wrap that one instead of introducing a new one :)

**Comment by Torxed** [comment]
I'm back from vacation starting today, this looks good and I'll catch up and merge things throughout the week.
People have been busy while I was cutting the grass and trimming hedges apparently ^^

**Comment by svartkanin** [comment]
@dylanmtaylor thanks for the feedback, I've updated the PR now to actually capture the mirror call error instead

**Review by JoelLarson — APPROVED** [review]

**Comment by Torxed** [comment]
It took 28 days, but I finally merged it! Thanks for improving this area, it's one of the features I use the most ^^

**Pull request merged**